zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -243,8 +335,40 @@ private[codec] object EncoderDecoder                   {
       } else if (jsonDecoders.length == 1) {
         jsonDecoders(0)(body).map { result => inputs(0) = result }
       } else {
-        ZIO.foreachDiscard(jsonDecoders.zipWithIndex) { case (decoder, index) =>
-          decoder(body).map { result => inputs(index) = result }
+        val indexByName = flattened.content.zipWithIndex.map { case (codec, idx) =>
+          codec.name.getOrElse("field" + idx.toString) -> idx
+        }.toMap
+        body.asMultipartFormStream.flatMap { form =>
+          form.fields.runForeach { field =>
+            indexByName.get(field.name) match {
+              case Some(idx) =>
+                flattened.content(idx) match {
+                  case BodyCodec.Multiple(schema, _, _) if schema == Schema[Byte] =>
+                    field match {
+                      case FormField.Binary(_, data, _, _, _)          =>
+                        ZIO.succeed {
```

Comment:
Since you are just doing `ZIO.suceed` for the side effect, I'd suggest, instead, to do the side-effect directly, and return `ZIO.unit`. This eliminates some overhead and allocations.